compiler/src/compiler/ast.rs

```diff
@@ -154,15 +156,15 @@ impl FindAst for Ast {
 }
 impl FindAst for Struct {
     fn find(&self, id: &Id) -> Option<&Ast> {
-        for (key, value) in &self.fields {
-            if let Some(ast) = key.find(id) {
-                return Some(ast);
-            }
-            if let Some(ast) = value.find(id) {
-                return Some(ast);
-            }
-        }
-        None
+        self.positional_fields
+            .iter()
+            .filter_map(|value| value.find(id))
+            .chain(
+                self.named_fields
+                    .iter()
+                    .filter_map(|(key, value)| key.find(id).or_else(|| value.find(id))),
+            )
+            .next()
```

Comment:
We can simplify this by first putting all values into an iterator and then trying to find the ID on that.

```suggestion
        self.positional_fields
            .iter()
            .chain(self.named_fields.iter().flat_map(|(key, value)| vec![key, value].iter()))
            .filter_map(|value| value.find(id))
            .next()
```

Not entirely sure about the ownership of the `vec` though